dotCMS/src/main/java/com/dotcms/storage/DataBaseStoragePersistenceAPIImpl.java

```diff
@@ -0,0 +1,652 @@
+        try (final OutputStream outputStream = Files.newOutputStream(file.toPath())) {
+            writerDelegate.write(outputStream, object);
+        }
+    }
+
+    @Override
+    public Future<Object> pushFileAsync(final String groupName, final String path, final File file,
+            final Map<String, Serializable> extraMeta) {
+
+        return DotConcurrentFactory.getInstance().getSubmitter(STORAGE_POOL).submit(
+                () -> this.pushFile(groupName, path, file, extraMeta)
+        );
+    }
+
+    @Override
+    public Future<Object> pushObjectAsync(final String bucketName, final String path,
+            final ObjectWriterDelegate writerDelegate,
+            final Serializable object, final Map<String, Serializable> extraMeta) {
+
+        return DotConcurrentFactory.getInstance().getSubmitter(STORAGE_POOL).submit(
+                () -> this.pushObject(bucketName, path, writerDelegate, object, extraMeta)
+        );
+    }
+
+    @Override
+    public File pullFile(final String groupName, final String path) throws DotDataException {
+        final String groupNameLC = groupName.toLowerCase();
+        final String pathLC = path.toLowerCase();
+        final MutableObject<File> file = new MutableObject<>(null);
+
+        try (Connection connection = getConnection()) {
+            final List<Map<String, Object>> storageResult = Try.of(() ->
+                    new DotConnect()
+                            .setSQL("SELECT hash FROM storage WHERE group_name = ? AND path = ?")
+                            .addParam(groupNameLC).addParam(pathLC)
+                            .loadObjectResults(connection)).getOrElse(Collections::emptyList);
+
+            if (!storageResult.isEmpty()) {
+
+                final Optional<Object> objectOpt = storageResult.stream()
+                        .map(map -> map.get("hash")).findFirst();
+
+                final String objectHash = (String) objectOpt.get();
+                if (UtilMethods.isSet(objectHash)) {
+                    file.setValue(this.createJoinFile(objectHash));
+                }
+            } else {
+                throw new DoesNotExistException(
+                        "The storage, group: " + groupName + ", path: " + path
+                                + " does not exists");
+            }
+
+        } catch (Exception e) {
+            Logger.error(DataBaseStoragePersistenceAPIImpl.class, e.getMessage(), e);
+            throw new DotDataException(e);
+        }
+
+        return file.getValue();
+    }
+
+    private File createJoinFile(final String hashId) throws IOException, DotDataException {
+
+        final File file = FileUtil.createTemporaryFile("dot-db-storage-recovery", ".tmp", true);
+        try (final FileJoiner fileJoiner = new FileJoiner(file)) {
+            final HashBuilder fileHashBuilder = Try.of(Hashing::sha256)
+                    .getOrElseThrow(DotRuntimeException::new);
+            final Connection connection = this.getConnection();
```

Comment:
Codacy found an issue: [Ensure that resources like this Connection object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)